worker/peergrouper/mock_test.go

```diff
@@ -479,21 +479,34 @@ func (session *fakeMongoSession) Set(members []replicaset.Member) error {
 }
 
 func (session *fakeMongoSession) StepDownPrimary() error {
-	logger.Debugf("StepDownPrimary")
 	if err := session.errors.errorFor("Session.StepDownPrimary"); err != nil {
+		logger.Debugf("StepDownPrimary error: %v", err)
 		return err
 	}
+	logger.Debugf("StepDownPrimary")
 	status := session.status.Get().(*replicaset.Status)
 	members := session.members.Get().([]replicaset.Member)
+	membersById := make(map[int]replicaset.Member, len(members))
+	for _, member := range members {
+		membersById[member.Id] = member
+	}
 	// We use a simple algorithm, find the primary, and all the secondaries that don't have 0 priority. And then pick a
 	// random secondary and swap their states
 	primaryIndex := -1
 	secondaryIndexes := []int{}
-	for i, member := range status.Members {
-		if member.State == replicaset.PrimaryState {
+	var info []string
+	for i, statusMember := range status.Members {
+		if statusMember.State == replicaset.PrimaryState {
 			primaryIndex = i
-		} else if member.State == replicaset.SecondaryState && (members[i].Priority == nil || *members[i].Priority > 0) {
-			secondaryIndexes = append(secondaryIndexes, i)
+			info = append(info, fmt.Sprintf("%d: current primary", statusMember.Id))
+		} else if statusMember.State == replicaset.SecondaryState {
+			confMember := membersById[statusMember.Id]
+			if confMember.Priority == nil || *confMember.Priority > 0 {
```

Comment:
Did you add _hasPriority_ as a convenience method with the intention of using it here?

Reply:
I started using it, and then realized I actually don't want it. I was confused because a test was failing in a non-obvious way (members list was not actually sorted).
I was intending to get rid of all the hasPriority stuff because it was actually a red-herring. (votes *is* always punned with Priority so we know the value of it, and don't have to worry about floating point printing, etc.)